### PR TITLE
Fix a few cert tests

### PR DESCRIPTION
**Pull request by sherryzy**
b/327656698

### CODE REVIEW

File: net/cert/pki/test_helpers.cc [hunk 1]
```diff
@@ -423,7 +423,11 @@ bool ReadVerifyCertChainTestFromFile(const std::string& file_path_ascii,
 std::string ReadTestFileToString(const std::string& file_path_ascii) {
   // Compute the full path, relative to the src/ directory.
   base::FilePath src_root;
+#if defined(STARBOARD)
+  base::PathService::Get(base::DIR_TEST_DATA, &src_root);
```
Comment (jellefoks) [inline net/cert/pki/test_helpers.cc:427]: Can we instead modify base/base_paths_starboard.cc to implement support for DIR_SOURCE_ROOT and return the same as DIR_TEST_DATA?

Comment (sherryzy) [reply net/cert/pki/test_helpers.cc:427]: Replied here: https://github.com/youtube/cobalt/pull/2616#discussion_r1529054434

Comment (sherryzy) [reply net/cert/pki/test_helpers.cc:427]: Actually, this is a good example that I shouldn't make this change. Replacing it with DIR_TEST_DATA breaks cert/internal/crl_unittest.cc. Will change this back.

**Review by johnxwork — COMMENTED** [review]
Some of the changes were copied from modern boringssl, right? Do we want to wait for the git update for them? Or are they already removed from upstream but we still need them here?

**Comment by sherryzy** [comment]
> Some of the changes were copied from modern boringssl, right? Do we want to wait for the git update for them? Or are they already removed from upstream but we still need them here?

Only third_party/boringssl/src/include/openssl/time.h is copied from modern boringssl. All the other related changes is just bring the original net code from m114 back. I'm not sure we should wait for an borringssl update, since it unblocks many cert/der features. Also it's just adding a header file which won't cause much merge conflicts when an update happens?

**Comment by johnxwork** [comment]
> > Some of the changes were copied from modern boringssl, right? Do we want to wait for the git update for them? Or are they already removed from upstream but we still need them here?
> 
> Only third_party/boringssl/src/include/openssl/time.h is copied from modern boringssl. All the other related changes is just bring the original net code from m114 back. I'm not sure we should wait for an borringssl update, since it unblocks many cert/der features. Also it's just adding a header file which won't cause much merge conflicts when an update happens?

I'm only confused by net/cert/pki/* which doesn't appear in upstream chromium code. I'm guessing this was moved to third_party/boringssl very recently. Anyway the impact is limited.

**Review by johnxwork — APPROVED** [review]

**Pull request merged**